node/src/components/chain_synchronizer/operations.rs

```diff
@@ -725,13 +725,30 @@ pub(super) async fn run_chain_sync_task(
 
     let maybe_last_emergency_restart_era_id = chainspec.protocol_config.last_emergency_restart;
     if let Some(last_emergency_restart_era) = maybe_last_emergency_restart_era_id {
-        if last_emergency_restart_era > trusted_block_header.era_id() {
+        // After an emergency restart, the old validators cannot be trusted anymore. So the last
+        // block before the restart or a later block must be given by the trusted hash. That way we
+        // never have to use the untrusted validators' finality signatures.
+        if trusted_block_header.next_block_era_id() < last_emergency_restart_era {
             return Err(Error::TryingToJoinBeforeLastEmergencyRestartEra {
                 last_emergency_restart_era,
                 trusted_hash,
                 trusted_block_header,
             });
         }
+        // If the trusted hash specifies the last block before the emergency restart, we have to
+        // compute the immediate switch block ourselves, since there's no other way to verify that
+        // block. We just sync the trie there and and return, so the upgrade can be applied.
```

Comment:
```suggestion
        // block. We just sync the trie there and return, so the upgrade can be applied.
```

Reply:
Done in https://github.com/casper-network/casper-node/pull/2580/commits/6f42ce72d59aece5f145be3d7b92264f1eb7efcf.